frontend/packages/dev-console/src/components/health-checks/AddHealthChecksForm.tsx

```diff
@@ -0,0 +1,82 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import * as yup from 'yup';
+import { Formik } from 'formik';
+import { FirehoseResult, LoadingBox, StatusBox, history } from '@console/internal/components/utils';
+import { K8sResourceKind, k8sUpdate, modelFor, referenceFor } from '@console/internal/module/k8s';
+import { getResourcesType } from '../edit-application/edit-application-utils';
+import { healthChecksProbesValidationSchema } from './health-checks-probe-validation-utils';
+import { getHealthChecksData } from './create-health-checks-probe-utils';
+import AddHealthChecks from './AddHealthChecks';
+import { updateHealthChecksProbe } from './health-checks-utils';
+
+type AddHealthChecksFormProps = {
+  resource?: FirehoseResult<K8sResourceKind>;
+  currentContainer: string;
+};
+
+const AddHealthChecksForm: React.FC<AddHealthChecksFormProps> = ({
+  resource,
+  currentContainer,
+}) => {
+  if (!resource.loaded && _.isEmpty(resource.loadError)) {
+    return <LoadingBox />;
+  }
+
+  if (resource.loadError) {
+    return <StatusBox loaded={resource.loaded} loadError={resource.loadError} />;
+  }
+
+  const container = _.find(
+    resource.data.spec.template.spec.containers,
+    (data) => data.name === currentContainer,
+  );
+
+  if (_.isEmpty(container)) {
+    return <div className="text-center">Container not found</div>;
+  }
+
+  const handleSubmit = (values, actions) => {
+    const updatedResource = updateHealthChecksProbe(values, resource.data, container);
+
+    k8sUpdate(modelFor(referenceFor(resource.data)), updatedResource)
+      .then(() => {
+        actions.setSubmitting(false);
+        actions.setStatus({ error: '' });
+        history.push(`/topology/ns/${updatedResource.metadata.namespace}`);
```

Comment:
We shouldn't be going back to topology always. We should return to the page the user was at before.
Keep in mind that in between the user may have changed containers so back in history doesn't work. Unless the dropdown just replaced the URL; i think it should replace instead of push.

Reply:
Fixed it. Instead of redirecting it to topology every time. now redirecting it back from where the user is landing to the health checks page.